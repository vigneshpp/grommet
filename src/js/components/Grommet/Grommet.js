import React, { Component } from 'react';
import { createGlobalStyle } from 'styled-components';

import {
  ResponsiveContext,
  ThemeContext,
  ContainerTargetContext,
} from '../../contexts';

import {
  backgroundIsDark,
  deepMerge,
  getBreakpoint,
  getDeviceBreakpoint,
  normalizeColor,
} from '../../utils';
import { base as baseTheme } from '../../themes';
import { StyledGrommet } from './StyledGrommet';

const FullGlobalStyle = createGlobalStyle`
  body { margin: 0; }
`;

class Grommet extends Component {
  static displayName = 'Grommet';

  static getDerivedStateFromProps(nextProps, prevState) {
    const {
      background: backgroundProp,
      dir,
      theme = {},
      themeMode,
    } = nextProps;
    const { theme: stateTheme, themeProp, themeModeProp } = prevState;

    const nextTheme = deepMerge(baseTheme, theme);
    if (!stateTheme || theme !== themeProp || themeMode !== themeModeProp) {
      const {
        colors: { background: themeBackground },
      } = nextTheme.global;

      // get initial value for dark so we can normalize background color
      nextTheme.dark = (themeMode || theme.defaultMode) === 'dark';
      const color = normalizeColor(
        backgroundProp || themeBackground,
        nextTheme,
      );

      // After normalizing, we set nextTheme.dark once more.
      // It is necessary that we set it twice. We have to handle two cases:
      // 1. Caller passes in a color object or a color name that resolves an
      //    object. In this case, we want to set dark as line 38 shows. The
      //    second set, in line 46, is a no-op.
      // 2. Caller passes a specific color value or a color name that resolves
      //    to a specific color value. In this case, we want dark to be set
      //    based on that color, which line 46 will do.
      // The double set of nextTheme.dark allows us to handle both cases here
      // without having to duplicate color object + name + dark mode detection
      // code here that is already in normalizeColor and backgroundIsDark.
      nextTheme.dark = backgroundIsDark(color, nextTheme);
      nextTheme.baseBackground = backgroundProp || themeBackground;

      if (dir) {
        nextTheme.dir = dir;
      }
      return {
        theme: nextTheme,
        themeProp: theme,
        themeModeProp: themeMode,
      };
    }

    return null;
  }

  state = {};

  componentDidMount() {
    window.addEventListener('resize', this.onResize);
    this.onResize();
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.onResize);
  }

  onResize = () => {
    const { theme, responsive } = this.state;

    let breakpoint;
    // responsive would be undefined in the case of SSR or initial page load
    if (!responsive) {
      // In the case of SSR we'll need to use the user agent breakpoint
      breakpoint = this.deviceResponsive();
    }
    // Initial page load where both responsive and breakpoint are undefined
    if (!breakpoint) breakpoint = getBreakpoint(window.innerWidth, theme);

    if (breakpoint !== responsive) {
      this.setState({ responsive: breakpoint });
    }
  };

  deviceResponsive() {
    const { userAgent } = this.props;
    const { theme } = this.state;

    /*
     * Regexes provided for mobile and tablet detection are meant to replace
     * a full-featured specific library due to contributing a considerable size
     * into the bundle.
     *
     * User agents found https://deviceatlas.com/blog/list-of-user-agent-strings
     */
    if (userAgent) {
      if (
        /(tablet|ipad|playbook|silk)|(android(?!.*mobile))/i.test(userAgent)
      ) {
        return getDeviceBreakpoint('tablet', theme);
      }
      if (/Mobile|iPhone|Android/.test(userAgent)) {
        return getDeviceBreakpoint('phone', theme);
      }
      return getDeviceBreakpoint('computer', theme);
    }
    return undefined;
  }

  render() {
    const {
      children,
      full,
      containerTarget = typeof document === 'object'
        ? document.body
        : undefined,
      ...rest
    } = this.props;
    delete rest.theme;
    const { theme, responsive: stateResponsive } = this.state;

    // Value from state should be correct once we resize
    // On first render we try to guess otherwise set the default as a tablet
    const responsive =
      stateResponsive ||
      this.deviceResponsive() ||
      theme.global.deviceBreakpoints.tablet;

    return (
      <ThemeContext.Provider value={theme}>
        <ResponsiveContext.Provider value={responsive}>
          <ContainerTargetContext.Provider value={containerTarget}>
            <StyledGrommet full={full} {...rest}>
              {children}
            </StyledGrommet>
            {full && <FullGlobalStyle />}
          </ContainerTargetContext.Provider>
        </ResponsiveContext.Provider>
      </ThemeContext.Provider>
    );
  }
}

let GrommetDoc;
if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  GrommetDoc = require('./doc').doc(Grommet);
}
const GrommetWrapper = GrommetDoc || Grommet;

export { GrommetWrapper as Grommet };
