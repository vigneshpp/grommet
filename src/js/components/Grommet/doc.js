import { describe, PropTypes } from 'react-desc';

import { backgroundDoc } from '../../utils/prop-types';
import { getAvailableAtBadge } from '../../utils/mixins';

export const doc = Grommet => {
  const DocumentedGrommet = describe(Grommet)
    .availableAt(getAvailableAtBadge('Grommet', 'Utilities'))
    .description('The top level Grommet container.')
    .usage(
      `import { Grommet } from 'grommet';
<Grommet>...</Grommet>`,
    )
    .intrinsicElement('div');

  DocumentedGrommet.propTypes = {
    background: backgroundDoc,
    dir: PropTypes.oneOf(['rtl']).description(
      'Layout direction for right to left contexts',
    ),
    full: PropTypes.bool
      .description('Whether to take the whole viewport.')
      .defaultValue(false),
    plain: PropTypes.bool
      .description(
        `Whether or not Grommet should apply a global font-family, font-size,
        and line-height.`,
      )
      .defaultValue(false),
    cssVars: PropTypes.bool
      .description('Whether to expose the css variables.')
      .defaultValue(false),
    theme: PropTypes.object.description(
      'Custom styles for Grommet app component.',
    ),
    themeMode: PropTypes.oneOf(['dark', 'light']).description(
      `Dark vs. light theme variation. Default is unspecified and left to
      theme.`,
    ),
    userAgent: PropTypes.string.description(
      `User agent used to detect the device width for setting the initial
      breakpoint.`,
    ),
    containerTarget: PropTypes.object.description(
      `The node where Drop and Layer containers are inserted. Defaults to
      document.body which is almost always the right choice. This is used
      for less common cases like rendering within an internal node (e.g.
      shadow root).`,
    ),
  };

  return DocumentedGrommet;
};

export const themeDoc = {
  'grommet.extend': {
    description: 'Any additional style for Grommet.',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
  'global.font.face': {
    description: 'Custom font face declaration',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
};
