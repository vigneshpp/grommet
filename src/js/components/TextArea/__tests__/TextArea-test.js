import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';
import 'jest-styled-components';
import { axe } from 'jest-axe';
import 'jest-axe/extend-expect';
import 'regenerator-runtime/runtime';

import { Grommet } from '../../Grommet';
import { TextArea } from '..';

describe('TextArea', () => {
  afterEach(cleanup);

  test('should not have accessibility violations', async () => {
    const { container } = render(
      <Grommet>
        <TextArea a11yTitle="test" id="item" name="item" />
      </Grommet>,
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  test('basic', () => {
    const component = renderer.create(
      <Grommet>
        <TextArea id="item" name="item" />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('placeholder', () => {
    const component = renderer.create(
      <Grommet>
        <TextArea id="item" name="item" placeholder="placeholder" />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('plain', () => {
    const component = renderer.create(
      <Grommet>
        <TextArea id="item" name="item" plain />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('disabled', () => {
    const component = renderer.create(
      <Grommet>
        <TextArea disabled id="item" name="item" plain />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('focusIndicator', () => {
    const component = renderer.create(
      <Grommet>
        <TextArea id="item" name="item" focusIndicator />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('fill', () => {
    const component = renderer.create(
      <Grommet>
        <TextArea id="item" name="item" fill />
      </Grommet>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  [true, false, 'horizontal', 'vertical'].forEach(resize => {
    test(`resize ${resize}`, () => {
      const component = renderer.create(
        <Grommet>
          <TextArea id="item" name="item" resize={resize} />
        </Grommet>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  ['small', 'medium', 'large'].forEach(size => {
    test(`size ${size}`, () => {
      const component = renderer.create(
        <Grommet>
          <TextArea id="item" name="item" size={size} />
        </Grommet>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Event tests', () => {
    afterEach(cleanup);
    const keyEvent = {
      key: 'Backspace',
      keyCode: 8,
      which: 8,
    };

    test(`onKeyDown`, () => {
      let capturedEvent = null;
      const callback = event => {
        const { key, keyCode, which } = event;
        capturedEvent = { key, keyCode, which };
      };

      const component = render(
        <Grommet>
          <TextArea
            id="item"
            name="item"
            placeholder="item"
            onKeyDown={callback}
          />
        </Grommet>,
      );

      const textArea = component.getByPlaceholderText('item');

      fireEvent.keyDown(textArea, keyEvent);

      expect(capturedEvent).toEqual(expect.objectContaining(keyEvent));
    });

    test(`onKeyUp`, () => {
      let capturedEvent = null;
      const callback = event => {
        const { key, keyCode, which } = event;
        capturedEvent = { key, keyCode, which };
      };

      const component = render(
        <Grommet>
          <TextArea
            id="item"
            name="item"
            placeholder="item"
            onKeyUp={callback}
          />
        </Grommet>,
      );

      const textArea = component.getByPlaceholderText('item');

      fireEvent.keyUp(textArea, keyEvent);

      expect(capturedEvent).toEqual(expect.objectContaining(keyEvent));
    });

    test('onFocus', () => {
      const onFocus = jest.fn();
      const { container, getByPlaceholderText } = render(
        <Grommet>
          <TextArea name="item" placeholder="item" onFocus={onFocus} />
        </Grommet>,
      );
      fireEvent.focus(getByPlaceholderText('item'));
      expect(container.firstChild).toMatchSnapshot();
      expect(onFocus).toHaveBeenCalledTimes(1);
    });

    test('onChange', () => {
      const onChange = jest.fn();
      const { getByPlaceholderText } = render(
        <Grommet>
          <TextArea name="item" placeholder="item" onChange={onChange} />
        </Grommet>,
      );
      const input = getByPlaceholderText('item');
      fireEvent.change(input, {
        target: { value: 'Test' },
      });
      expect(input.value).toEqual('Test');
      expect(onChange).toHaveBeenCalledTimes(1);
    });

    test('onBlur is being called', () => {
      const onBlur = jest.fn();
      const { getByPlaceholderText } = render(
        <Grommet>
          <TextArea name="item" placeholder="item" onBlur={onBlur} />
        </Grommet>,
      );
      fireEvent.blur(getByPlaceholderText('item'));
      expect(onBlur).toHaveBeenCalledTimes(1);
    });
  });
});
