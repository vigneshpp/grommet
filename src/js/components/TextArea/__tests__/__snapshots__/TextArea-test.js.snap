// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Event tests onFocus 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
}

.c1 > circle,
.c1 > ellipse,
.c1 > line,
.c1 > path,
.c1 > polygon,
.c1 > polyline,
.c1 > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c1::-moz-focus-inner {
  border: 0;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

<div
  class="c0"
>
  <textarea
    class="c1"
    name="item"
    placeholder="item"
  />
</div>
`;

exports[`TextArea basic 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea disabled 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  outline: none;
  border: none;
  width: 100%;
  -webkit-appearance: none;
  opacity: 0.3;
  cursor: default;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    disabled={true}
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea fill 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  height: 100%;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea focusIndicator 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea placeholder 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="placeholder"
  />
</div>
`;

exports[`TextArea plain 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  outline: none;
  border: none;
  width: 100%;
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea resize false 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  resize: none;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea resize horizontal 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  resize: horizontal;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea resize true 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  resize: both;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea resize vertical 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  resize: vertical;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea size large 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  font-size: 22px;
  line-height: 28px;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    size="large"
  />
</div>
`;

exports[`TextArea size medium 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  font-size: 18px;
  line-height: 24px;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    size="medium"
  />
</div>
`;

exports[`TextArea size small 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  font-size: 14px;
  line-height: 20px;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    size="small"
  />
</div>
`;
